File: OIAJ/cafes1.java
import java.lang.*;
import java.util.*;

public class cafes1 {
    public static final int INF = 1000000000;
    public static String cafes(ArrayList<String> nombres, ArrayList<Integer> precios, ArrayList<Integer> distancias) {
        int n = nombres.size();
        int mini = INF;
        int minip = INF;
        int idx = -1;
        for (int i = 0; i < n; i++) {
            int prod = precios.get(i) * distancias.get(i);
            if (prod < mini) {
                mini = prod;
                minip = precios.get(i);
                idx = i;
            }
            else if (prod == mini && precios.get(i) < minip) {
                minip = precios.get(i);
                idx = i;
            }
        }
        return nombres.get(idx);
    }
}

File: OIAJ/cafes1.py
INF = 1000000000

def cafes(nombres, precios, distancias):
    N = len(nombres)
    mini = INF
    minip = INF
    idx = -1
    for i in range(N):
        c = precios[i] * distancias[i]
        if c < mini:
            mini = c
            minip = precios[i]
            idx = i
        elif c == mini and precios[i] < minip:
            minip = precios[i]
            idx = i

    return nombres[idx]
